src/Vector.js

```diff
@@ -16,63 +16,51 @@ class Vector
 	 */
 	static Create(str)
 	{
-		const allVectors = [];
-		const substrings = str.split(";");
+		const vectors = [];
 		
-		/* Helper function */
-		function CrossProduct(arr1, arr2)
-		{
-			const crossProduct = [];
-			for (let i = 0; i < arr1.length; i++)
-			{
-				for (let j = 0; j < arr2.length; j++)
-				{
-					crossProduct.push([arr1[i], arr2[j]]);
-				}
-			}
-			return crossProduct;
-		}
-		
-		substrings.forEach((substring) => {
+		/* Iterate over each vector string, delimited by semicolon */
+		str.split(";").forEach((substring) => {
 			const vectorString = substring.trim();
-			if (vectorString == "")
+			if (vectorString === "")
 			{
 				/* Ignore empty strings - common if vector list ended with semicolon */
 				return;
 			}
 			
-			const checkValid = vectorString.match(/\(-?\d+[\d{}+jhpmd]*, -?\d+[\d{}+jhpmd]*\)[\d{}+jhpmd]*/g);
-			if (checkValid == undefined || checkValid.length <= 0)
+			const checkValid = vectorString.match(/\ *\(\ *-?\d+[\d{}+jhpmd]*\ *,\ *-?\d+[\d{}+jhpmd]*\ *\)[\d{}+jhpmd]*\ */g);
```

Comment:
Still don't need the first or last. Again, the string is trimmed.
```suggestion
			const checkValid = vectorString.match(/\(\ *-?\d+[\d{}+jhpmd]*\ *,\ *-?\d+[\d{}+jhpmd]*\ *\)[\d{}+jhpmd]*/g);
```